mono/metadata/reflection.c

```diff
@@ -8779,8 +8779,16 @@ mono_reflection_create_custom_attr_data_args (MonoImage *image, MonoMethod *meth
 			return;
 		}
 
-		obj = (MonoObject *)(type_is_reference (mono_method_signature (method)->params [i]) ?
-			val : mono_value_box (domain, mono_class_from_mono_type (mono_method_signature (method)->params [i]), val));
+		if (type_is_reference (mono_method_signature (method)->params [i])) {
+			obj = (MonoObject*) val;
+		} else {
+			obj = mono_value_box_checked (domain, mono_class_from_mono_type (mono_method_signature (method)->params [i]), val, error);
+			if (!mono_error_ok (error)) {
+				g_free (val);
```

Comment:
Error handling in this function is wrong.

It's ignoring the fail label and, thus, not freeing arginfo and NULL'ing named_arg_info.